### PR TITLE
Fix/fesom hlrn

### LINKED ISSUE #903: Installing esm_tools on HLRN
I installed the esm_tools on HLRN. It seems to have worked, as I can call on esm_master, and esm_tools --version, however I got the following error message:
```
ERROR: flair 0.6.1 requires janome, which is not installed.
ERROR: flair 0.6.1 requires konoha<5.0.0,>=4.0.0, which is not installed.
ERROR: flair 0.6.1 has requirement mpld3==0.3, but you'll have mpld3 0.5.1 which is incompatible.
ERROR: flair 0.6.1 has requirement sentencepiece!=0.1.92, but you'll have sentencepiece 0.1.92 which is incompatible.
```
Will this be a problem, or can I just ignore it? 

**What I did:**
I followed the instructions on the main page.
I added the following to .bashrc:
```
module load git
module load anaconda3

export PATH=$PATH:~/.local/bin
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
```
And then:
```
$ source ~/.bashrc
$ git clone https://github.com/esm-tools/esm_tools.git
```
And then tried to install:
```
$ cd esm_tools
$ ./install.sh
```
**System info**
 - Supercomputer: HLRN (blogin.hlrn.de)
 - Version: 6.19.5

### PR DESCRIPTION
**Pull request by mandresm**
Cleanups for fesom version 1.4 and 2 in HLRN

Closes #903

### CODE REVIEW

File: configs/components/fesom/fesom-2.0.yaml [hunk 1]
```diff
@@ -336,15 +327,6 @@ config_sources:
...
 
-compiletime_environment_changes:
-        add_export_vars:
-                taken2from:      fesom2_compile
-
-runtime_environment_changes:
-        add_export_vars:
-                takenfrom:      fesom2_run
-                taken2from:     fesom2_ru
-
```
Comment (pgierz) [inline configs/components/fesom/fesom-2.0.yaml:347]: Thank you! These have been annoying me for a while

File: src/esm_tests/test_utilities.py [hunk 2]
```diff
@@ -261,6 +261,7 @@ def sort_dict(dict_to_sort):
         Dictionary sorted.
     """
     if isinstance(dict_to_sort, dict):
+        dict_to_sort = {str(key): value for key, value in dict_to_sort.items()}
```
Comment (pgierz) [inline src/esm_tests/test_utilities.py:264]: I'm not 100% sure this function will actually work for older Python version. Python dictionaries used to be unsorted by nature, only newer than 3.7 was order guaranteed to be kept same as insertion order (see here: https://realpython.com/sort-python-dictionary/#rediscovering-dictionary-order-in-python)

Not a problem with the code, just a "learn more about Python" moment 😉

**Review by pgierz — APPROVED** [review]
Looks good to me!

File: src/esm_tests/test_utilities.py (hunk 2 above)
Comment (mandresm) [reply src/esm_tests/test_utilities.py:264]: Thanks, good to know! :) Luckily this is only to make things more beautiful for the esm_tests output.

**Comment by mandresm** [comment]
#approve-changes

**Comment by mandresm** [comment]
#bump

**Pull request merged**